File: 3.1P Clock Class/Clock/Clock.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Clock
{
    public class Clock
    {
        private Counter _hour;
        private Counter _minute;
        private Counter _second;

        public Clock()
        {
            _hour = new Counter("Hours");
            _minute = new Counter("Minutes");
            _second = new Counter("Seconds");
        }

        public void Ticks()
        {
            _second.Increment();
            if (_second.Ticks == 60)
            {
                _minute.Increment();
                _second.Reset();
            }
            if (_minute.Ticks == 60)
            {
                _hour.Increment();
                _minute.Reset();
            }
            if (_hour.Ticks == 24)
            {
                _hour.Reset();
            }
        }

        public void Reset() 
        {
            _hour.Reset();
            _minute.Reset();
            _second.Reset();
        }

        public string DisplayTime()
        {            
            return String.Format("{0:D2}:{1:D2}:{2:D2}", _hour.Ticks, _minute.Ticks, _second.Ticks);            
        }


    }
}

File: 11.1P Clock in Python/ClockInPy/Clock.py
from Counter import Counter #importing the Counter class from the Counter module

class Clock:
    #self. is used to refer to the Clock's instances
    def __init__(self): #constructor
        self.hour = Counter("Hours")
        self.minute = Counter("Minutes")
        self.second = Counter("Seconds")

    def ticks(self):
        self.second.increment()
        
        if (self.second.count == 60):
            self.minute.increment()
            self.second.reset()
            
        if (self.minute.count == 60):
            self.hour.increment()
            self.minute.reset()
            
        if (self.hour.count == 24):
            self.hour.reset()

    def reset(self): #reset the value of hour, minute and second
        self.hour.reset()
        self.minute.reset()
        self.second.reset()

       
    def display_time(self):  #display the current time in HH:MM:SS format
        return "{:02d}:{:02d}:{:02d}".format(self.hour.count, self.minute.count, self.second.count)
